File: smoke/smokeTest-mysql/python/smokeTest-my.py
import pymysql
import sys


def myconn(host, port, sql, opt):
    conn = pymysql.connect(host=host, port=port , database='postgres', user='abc', password='abc')
    cursor = conn.cursor()
    cursor.execute(sql)
    if opt == 'y':
        res = cursor.fetchall()
        print(str(res) + '\n')
    conn.commit()
    cursor.close()
    conn.close()

def myTest(host, port):
    sql = ['drop table if exists mypython_sm;', 
            'create table mypython_sm(a int primary key, b text);',
            "insert into mypython_sm values(1, 'abc'),(2, 'bcd'),(3, 'cde');", 
            "update mypython_sm set b = 'bcd' where a = 1;", 
            "delete from mypython_sm where a = 1;"
            ]

    select = "\nselect * from mypython_sm;"

    num = 0
    for sqls in sql:
        print('========' + sqls + '========')
        myconn(host, int(port), sqls, 'n')
        print('success')

        if num > 1 and num < 5:
            print(select)
            myconn(host, int(port), select, 'y')
        
        num = num + 1

if __name__ == '__main__':
    myTest(sys.argv[1], sys.argv[2])

File: smoke/smokeTest-mysql/js/smoketest-my.js
// 使用process获取命令行传的参数
var arguments = process.argv.splice(2);
var hosts = arguments[0];
var ports  = arguments[1];

var mysql  = require('mysql');

var connection = mysql.createConnection({
    host     : hosts,
    user     : 'abc',
    password : 'abc',
    port: ports,
    database: 'postgres'
});
//console.log(connection)


connection.connect();

var sql1 = 'drop table if exists myjs_sm';
var sql2 = 'create table if not exists myjs_sm(a int primary key,b text);'
var sql3 = 'insert into myjs_sm values(1,"abc")';
var sqls = 'select * from myjs_sm';
var sql4 = 'update myjs_sm set b = "asd" where a = 1';
var sql5 = 'delete from myjs_sm where a = 1';

var  execsql=function(arg1){
//查
	connection.query(arg1, function (err, result) {
    		if(err){
        	console.log(err.message);
        	return;
    	}

    	console.log('-----------------------[ "' + arg1 + '" ]');
    	console.log(result);
    	console.log('------------------------------------------------------------\n\n');
	});

	//connection.end();
}

execsql(sql1);
execsql(sql2);
execsql(sql3);
execsql(sqls);
execsql(sql4);
execsql(sqls);
execsql(sql5);
connection.end();
